### PR TITLE
update the Resolver api to use the new package config api

### LINKED ISSUE #459: Migrate to the latest package_config package
~This is needed to unblock package:test users~ Note that this is no longer blocking as we are doing a non-breaking release of package_config. But we should still update once it is available.

### PR DESCRIPTION
**Pull request by jakemac53**
Closes https://github.com/dart-lang/tools/issues/459.

Blocked on https://github.com/dart-lang/package_config/issues/68 and a published version of package_config.

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,15 +1,15 @@
 name: coverage
-version: 0.13.6
+version: 0.13.7
```
Comment (grouma) [inline pubspec.yaml]: [optional] Should we just bump this to 1.0.0?

**Review by grouma — APPROVED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (jakemac53) [reply pubspec.yaml]: this isn't breaking - I would do it separately afterwords or wait for NNBD?

Comment (grouma) [reply pubspec.yaml]: Ah true. Let's just wait until NNBD.

**Review by natebosch — APPROVED** [review]

**Comment by jakemac53** [comment]
PTAL, updated to rely on the new published version and fixed .packages parsing

**Review by cbracken — APPROVED** [review]
We should probably add a test here.

**Comment by jakemac53** [comment]
Yep I am working on adding tests for a resolver with the new format - which also brought up an unexpected issue where the new package_config requires absolute file uris :D. Hooray for coverage :fireworks: .

**Comment by cbracken** [comment]
You have all the LGTMs I have to give. Or at least all that matter per CL :)

**Comment by jakemac53** [comment]
Going to do a final check through an internal global before landing

**Comment by jakemac53** [comment]
Of course I have to land package_config internally first so this will be delayed likely until monday

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+## 0.13.7 - 2020-02-14
+
+* Update to package_config `1.9.0` which supports package_config.json
```
Comment (kevmoo) [inline CHANGELOG.md]: Note that `packageRoot` is now deprecated in the API?

Comment (jakemac53) [reply CHANGELOG.md]: done

**Pull request merged**